cmd/crdCreator/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const baseDir string = "shim/crds"
+
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;watch;list;create
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	ctx := context.Background()
+
+	log.Info("Setting up k8s client")
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	crdFiles := []string{
+		"storageclusters.ocs.openshift.io.yaml",
+		"ocsinitializations.ocs.openshift.io.yaml",
+		"noobaas.noobaa.io.yaml",
+		"objectbucketclaims.objectbucket.io.yaml",
+	}
+
+	for _, f := range crdFiles {
+		file, err := os.Open(fmt.Sprintf("%s/%s", baseDir, f))
+		if err != nil {
+			log.Error(err, "unable to read crd templates")
+			os.Exit(1)
+		}
+		crd := &v1.CustomResourceDefinition{}
+		decoder := yaml.NewYAMLToJSONDecoder(file)
+		_ = decoder.Decode(crd)
+		if err := k8sClient.Create(ctx, crd); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
- `Create` on a kubeclient is similar to, get a mapper, init a discovery, infer the traits of this resource and perform this op
- My previous comment is to use typed client, which just performs this op, any reason in not using it?